recipes/full_finetune_distributed.py

```diff
@@ -563,6 +584,12 @@ def recipe_main(cfg: DictConfig) -> None:
         )
 
     init_process_group(backend="gloo" if cfg.device == "cpu" else "nccl")
+    if cfg.get("fsdp_cpu_offload", False):
```

Comment:
Again not a fan of this being in the recipe code, this should likely be in a utility or something which the user doesn't probably need to reason about.

cc: @ebsmothers to help brainstorm this a bit

Reply:
Yeah +1 can we put this in a utility?

Reply:
Yeah, definitely!

Reply:
Bumping this comment. Also I don't fully understand the code comment just below: do we only need to `set_num_threads` in the case of fused optimizer? Looking at the answer.ai code seems like the answer is no, but based on the code comment it's not clear to me

Reply:
We see the biggest gain setting this when fused optimizer is used as it controls threads used for intra-op parallelism, and CPU optimizer is the heaviest CPU op. We may have very slight speed ups when not using fused optimizer, but those can be investigated separately of this PR.